src/main/scala/io/iohk/ethereum/domain/SignedTransaction.scala

```diff
@@ -55,15 +53,15 @@ object SignedTransaction {
     }
   }
 
-  private def getSender(tx: Transaction, signature: ECDSASignature, chainId: Byte): Option[Address] = {
-    val ECDSASignature(_, _, v) = signature
+  def getSender(tx: SignedTransaction): Option[Address] = {
+    val ECDSASignature(_, _, v) = tx.signature
     val bytesToSign: Array[Byte] = if (v == ECDSASignature.negativePointSign || v == ECDSASignature.positivePointSign) {
-      generalTransactionBytes(tx)
+      generalTransactionBytes(tx.tx)
     } else {
-      chainSpecificTransactionBytes(tx, chainId)
+      chainSpecificTransactionBytes(tx.tx, chainId)
     }
 
-    val recoveredPublicKey: Option[Array[Byte]] = signature.publicKey(bytesToSign, Some(chainId))
+    val recoveredPublicKey: Option[Array[Byte]] = tx.signature.publicKey(bytesToSign, Some(chainId))
```

Comment:
you could replace this `tx.signature` with `v` extracted before, but consider change of the name for something more descriptive

Reply:
not sure what you mean, `v` is only a part of signature, and to calculate public key i need whole signature.

Reply:
yup, my mistake, ignore this comment